docs/scripts/formattedTSDemos.js

```diff
@@ -1,55 +1,100 @@
+/* eslint-disable no-console */
+/* eslint-disable no-continue */
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-restricted-syntax */
+
 /**
  * Transpiles and formats TS demos.
  * Can be used to verify that JS and TS demos are equivalent. No introduced change
  * would indicate equivalence.
  */
-const childProcess = require('child_process');
+
+/**
+ * List of demos to ignore when transpiling
+ * Example: "app-bar/BottomAppBar.tsx"
+ */
+const ignoreList = [];
+
 const fse = require('fs-extra');
 const path = require('path');
+const babel = require('@babel/core');
 const prettier = require('prettier');
-const util = require('util');
+const os = require('os');
+
+const babelConfig = {
+  presets: ['@babel/preset-typescript'],
+  plugins: ['unwrap-createStyles'],
+  generatorOpts: { retainLines: true },
+  babelrc: false,
+  configFile: false,
+};
+
+const prettierConfig = prettier.resolveConfig(process.cwd(), {
+  config: path.join(__dirname, '../../prettier.config.js'),
+});
+
+const watchMode = process.argv.some(arg => arg === '--watch');
 
-const exec = util.promisify(childProcess.exec);
+async function getFiles(root) {
+  const files = [];
+  const folderItem = await fse.readdir(root);
 
-async function getUnstagedGitFiles() {
-  const { stdout } = await exec('git diff --name-only');
-  const list = stdout.trim();
+  for (const name of folderItem) {
+    const filePath = path.join(root, name);
+    const stat = await fse.stat(filePath);
 
-  if (list === '') {
-    // "".split(" ") => [""]
-    return [];
+    if (stat.isFile()) {
+      if (!filePath.endsWith('.tsx')) continue;
+      if (ignoreList.some(ignorePath => filePath.endsWith(path.normalize(ignorePath)))) continue;
+      files.push(filePath);
+    } else if (stat.isDirectory()) {
+      files.push(...(await getFiles(filePath)));
+    }
   }
 
-  return list.split('\n');
+  return files;
 }
 
 function fixBabelGeneratorIssues(source) {
-  return source.replace(/\/>\n\n/g, '/>\n').replace(/,\n\n/g, ',\n');
+  return source
+    .replace(/\/>(\r\n|\r|\n)(\r\n|\r|\n)/g, `/>${os.EOL}`)
```

Comment:
```suggestion
    .replace(/\/>(\r*\n){2}/g, `/>${os.EOL}`)
```

Reply:
There are systems that use just `\r` (granted they're really old), should we ignore those?

Reply:
Used `(\r?\n){2}` instead, seems git only supports `\n` and `\r\n`